source/Ship.cpp

```diff
@@ -275,29 +275,61 @@ void Ship::Load(const DataNode &node)
 				if(child.Size() >= 2)
 					outfit = GameData::Outfits().Get(child.Token(1));
 			}
-			Angle gunPortAngle = Angle(0.);
-			bool gunPortParallel = false;
+			Hardpoint::BaseAttributes attributes;
+			attributes.baseAngle = Angle(0.);
+			attributes.isParallel = false;
+			attributes.isOmnidirectional = true;
 			bool drawUnder = (key == "gun");
 			if(child.HasChildren())
 			{
+				bool defaultBaseAngle = true;
 				for(const DataNode &grand : child)
 				{
+					bool needToCheckAngles = false;
 					if(grand.Token(0) == "angle" && grand.Size() >= 2)
-						gunPortAngle = grand.Value(1);
+					{
+						attributes.baseAngle = grand.Value(1);
+						needToCheckAngles = true;
+						defaultBaseAngle = false;
+					}
 					else if(grand.Token(0) == "parallel")
-						gunPortParallel = true;
+						attributes.isParallel = true;
+					else if(grand.Token(0) == "arc" && grand.Size() >= 3)
+					{
+						attributes.isOmnidirectional = false;
+						attributes.minArc = Angle(grand.Value(1));
+						attributes.maxArc = Angle(grand.Value(2));
+						needToCheckAngles = true;
+						if(attributes.minArc.Degrees() > attributes.maxArc.Degrees())
+						{
+							grand.PrintTrace("Warning: Minimum arc is higher than maximum arc. Might not work as expected.");
+						}
+					}
 					else if(grand.Token(0) == "under")
 						drawUnder = true;
 					else if(grand.Token(0) == "over")
 						drawUnder = false;
 					else
-						grand.PrintTrace("Skipping unrecognized attribute:");
+						grand.PrintTrace("Warning: Child nodes of \"" + key
+							+ "\" tokens can only be \"angle\", \"parallel\", or \"arc\":");
+
+					if(needToCheckAngles && !defaultBaseAngle && !attributes.isOmnidirectional)
+					{
+						attributes.minArc += attributes.baseAngle;
+						attributes.maxArc += attributes.baseAngle;
+					}
+				}
+				if(!attributes.isOmnidirectional && defaultBaseAngle)
```

Comment:
If you don't specify the base angle, the min and max arcs look like absolute angles. Is it your intention?

Reply:
Yeah, it seems to work fine so I leave it as that.